File: 백준/Bronze/2884. 알람 시계/알람 시계.py
var = str(input())
var = var.split()

match (int(var[0]) == 0, int(var[-1]) - 45 >= 0):
    case (True, True):
        print('{} {}'.format(var[0], int(var[-1]) - 45))
    case (True, False):
        minus_var = int(var[-1]) - 45
        print('23 {}'.format(60 + minus_var))
    case (False, True):
        print('{} {}'.format(var[0], int(var[-1]) - 45))
    case (False, False):
        minus_var = int(var[-1]) - 45
        print('{} {}'.format(int(var[0]) - 1, 60 + minus_var))

File: node.js/백준/Bronze/2884. 알람 시계/알람 시계.js
let rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

let input;

rl.on('line', function(li) {
    input = li.split(' ');
    rl.close();
}).on('close', function() {
    let a = parseInt(input[0]);
    let b = parseInt(input[1]);
    time(a, b);
    process.exit();
})

function time(a, b) {
    b = b - 45;
    if (b < 0) {
        a = a - 1;
        b = 60 + b;
    }
    a = (a < 0)? 24 + a : a;
    console.log(a, b);
}